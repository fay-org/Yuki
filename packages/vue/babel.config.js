module.exports = {
  presets: ['@fect-ui/cli/preset'],
}
