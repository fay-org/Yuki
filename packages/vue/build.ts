import { BuildTaskConfig, TASK_NAME, runTask, commonOutput } from 'internal'
import { createBundle } from 'no-bump'

const { build } = createBundle({})

const commonInput = 'src/index.ts'

/**
 * I decide split the build task into four.
 * First is only bundle script module.
 * Second part only parser style moudle.
 * Thrid part will generator all analyzed and generator style deps graph.
 * Fourth part will generator the entry file as umd.
 *
 * The first task will have two sub task. with esm task coomonjs task
 */
