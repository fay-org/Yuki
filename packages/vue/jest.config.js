module.exports = {
  testPathIgnorePatterns: ['/node_modules/', '/dist/', '/lib/', '/es/'],
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx,vue}',
    '!src/utils/**',
    '!src/index.ts',
    '!src/**/type.ts',
    '!src/**/style.ts',
    '!src/**/interface.ts',
    '!src/back-top/**'
  ],
  collectCoverage: true,
  preset: '@fect-ui/cli/lib/config',
  testTimeout: 20000
}
