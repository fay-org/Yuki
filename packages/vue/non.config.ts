import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  lib: {
    input: path.join(__dirname, 'components'),
    format: ['cjs', 'es', 'umd'],
    name: 'fect'
  }
})
