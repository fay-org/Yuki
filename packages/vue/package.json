{
  "name": "@fect-ui/vue",
  "version": "1.4.0-rc.1",
  "description": "Minimalist UI components built on Vue",
  "main": "lib/index.js",
  "module": "es/index.js",
  "types": "types/index.d.ts",
  "jsdelivr": "lib/fect.min.umd.js",
  "scripts": {
    "collect": "non collect",
    "build": "npm run collect && non compile && non volar && node ./typing-template.js",
    "test": "non test"
  },
  "keywords": [
    "vue component",
    "fect",
    "vue ui"
  ],
  "homepage": "https://github.com/fect-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fect-org/fect"
  },
  "files": [
    "lib/",
    "es/",
    "types/",
    "vetur/",
    "README.md",
    "components.d.ts"
  ],
  "author": "XeryYue",
  "license": "MIT",
  "devDependencies": {
    "@fect-ui/cli": "^1.0.0"
  },
  "sideEffects": [
    "es/**/style/*",
    "lib/**/style/*",
    "*.css"
  ],
  "dependencies": {
    "@fect-ui/themes": "^2.2.0",
    "@fect-ui/vue-hooks": "^1.2.0",
    "proy": "^0.0.3",
    "@popperjs/core": "^2.11.2"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  }
}
