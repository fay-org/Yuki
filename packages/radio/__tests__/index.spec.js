import { mount } from '@vue/test-utils'
import { ref } from 'vue'
import Radio from '../index'

describe('Radio', () => {
  it('should be render as element', () => {
    const radio = mount(Radio, { props: { value: 'test' } })
    expect(radio.html()).toMatchSnapshot()
    expect(() => radio.unmount()).not.toThrow()
  })

  it('should be support disabled', () => {
    const wrapper = mount(<Radio value={'test'} disabled />)

    expect(wrapper.props('disabled')).toBeTruthy()
  })

  it('should be support different sizes', () => {
    const wrapper = mount({
      setup() {
        return { sizeList: ['mini', 'small', 'medium', 'large'] }
      },
      render() {
        return (
          <>
            {this.sizeList.map((item, i) => (
              <Radio size={item} key={i} value={item} />
            ))}
          </>
        )
      },
    })

    const els = wrapper.findAll('.fect-radio')
    expect(els[0].attributes('style')).toBe('--radioSize: 12px;')
    expect(els[1].attributes('style')).toBe('--radioSize: 14px;')
    expect(els[2].attributes('style')).toBe('--radioSize: 16px;')
    expect(els[3].attributes('style')).toBe('--radioSize: 18px;')
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be support cheked', () => {
    const wrapper = mount(<Radio value={'test'} checked={true} />)
    expect(wrapper.html()).toContain(
      '<span class="fect-radio__point  active"></span>',
    )
  })

  it('should  emit change event', async () => {
    const wrapper = mount({
      setup() {
        return { state: ref(0) }
      },
      render() {
        return (
          <Radio value={this.state} onChange={() => (this.state = 1)}>
            state: {this.state}
          </Radio>
        )
      },
    })

    wrapper.trigger('change')
    expect(wrapper.emitted('change').length).toEqual(1)
    expect(wrapper.html()).toContain('state: 0')
    await wrapper.find('[type="radio"]').trigger('change')
    expect(wrapper.html()).toContain('state: 1')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
