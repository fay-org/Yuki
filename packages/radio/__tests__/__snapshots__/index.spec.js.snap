// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should  emit change event 1`] = `<div class="fect-radio" style="--radioSize: 16px;"><label class=""><input type="radio" value="0"><span class="fect-radio__name"><span class="fect-radio__point  active"></span>state: 1</span></label></div>`;

exports[`Radio should be render as element 1`] = `
<div class="fect-radio" style="--radioSize: 16px;"><label class=""><input type="radio" value="test"><span class="fect-radio__name"><span class="fect-radio__point  "></span>
    <!----></span>
  </label></div>
`;

exports[`Radio should be support different sizes 1`] = `
<div class="fect-radio" style="--radioSize: 12px;"><label class=""><input type="radio" value="mini"><span class="fect-radio__name"><span class="fect-radio__point  "></span>
    <!----></span>
  </label></div>
<div class="fect-radio" style="--radioSize: 14px;"><label class=""><input type="radio" value="small"><span class="fect-radio__name"><span class="fect-radio__point  "></span>
    <!----></span>
  </label></div>
<div class="fect-radio" style="--radioSize: 16px;"><label class=""><input type="radio" value="medium"><span class="fect-radio__name"><span class="fect-radio__point  "></span>
    <!----></span>
  </label></div>
<div class="fect-radio" style="--radioSize: 18px;"><label class=""><input type="radio" value="large"><span class="fect-radio__name"><span class="fect-radio__point  "></span>
    <!----></span>
  </label></div>
`;
